Fail on duplicates when creating archives by default (#10143)
gradle/gradle: subprojects/docs/src/docs/userguide/upgrading_version_5.adoc

> // Copyright 2019 the original author or authors.

Author: I think this should be 2019

==== Local build cache type shouldn't be specified

> When configuring the local build cache the use of `BuildCacheConfiguration.local(Class)` and `local(Class, Action)` have now been deprecated; use `getLocal()` or `local(Action)` instead.

Author: @lptr, is this correct?

Author: Reverted for now

If your plugin/build only uses these types (and does not extend them), nothing has changed.

> ==== Archives fail on duplicates

Reviewer: ```suggestion
==== Archive tasks fail on duplicate files
```

> Until now archives defaulted to the `INHERIT` duplicates strategy, allowing the same path to exist multiple times in an archive.

Reviewer: I think we need to be more specific what `archives` means here. I think we should reference `archive tasks`, and maybe point to `Zip` and `Tar` specifically.

Author: The previous header already defines it:
> ==== Change in behavior for tasks extending `AbstractArchiveTask` or subtypes (`Zip`, `Jar`, `War`, `Ear`, `Tar`)

Reviewer: ```suggestion
Until now archive tasks defaulted to the `INCLUDE` duplicates strategy, allowing the same path to exist multiple times in an archive.
```

I don't think people read the previous header. Let's repeat the affected tasks again here. Or have one section (changes for archive tasks) with two sub-sections.

Author: Done

In Gradle 6.0 we are switching to `FAIL`, prohibiting duplicate files in archives.
If you still want to allow them, you can be specify that explicitly:

```
task archive(type: Zip) {
    duplicatesStrategy = DuplicatesStrategy.INCLUDE // allow duplicates
    archiveName = 'archive.zip'
    from 'src'
}
```

> *Note* that `Copy` and `Sync` tasks are unaffected: they still use the `INHERIT` duplicates strategy as default.

Reviewer: ```suggestion
*Note* that `Copy` and `Sync` tasks are unaffected: they still use the `INCLUDE` duplicates strategy as default.
```

Author: Meaning that inherit will delegate to include?
I.e. https://github.com/gradle/gradle/blob/623062702ba0bd809a0393cd53bed37f3ff9e58e/subprojects/core/src/main/java/org/gradle/api/internal/file/copy/DefaultCopySpec.java#L75
